### PR TITLE
Prototype hack involving default pictures for users with no picture

### PR DESCRIPTION
**Pull request by andrewconner**
@stkem @leogrim 

Getting user picture closer to a state where we can use the existing `ProcessedImages` pipeline. A roadblock is `pictureName` being a bad API. We'd like to move to using real, absolute URLs for pictures. This is testing that we don't break clients if I put in a backwards compatibility hack.

Shouldn't affect anyone but me, Eishay, and three test users.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala [hunk 1]
```diff
@@ -72,19 +73,44 @@ object BasicUser {
...
+  }
+
   def fromUser(user: User): BasicUser = {
     BasicUser(
       externalId = user.externalId,
       firstName = user.firstName,
       lastName = user.lastName,
-      pictureName = user.pictureName.getOrElse(S3UserPictureConfig.defaultName) + ".jpg",
+      pictureName = userToPictureName(user),
       username = user.username
```
Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala:107]: Do you plan on introducing new fields and deprecate this one?

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala:107]: Introducing new fields

Comment (leogrim) [inline kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala:107]: :+1:

**Comment by stkem** [comment]
looks save to me

**Pull request merged**